packages/react/src/components/AccountSettings/ChangePassword/ChangePassword.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+
+import { Logger } from 'aws-amplify';
+import { changePassword, translate } from '@aws-amplify/ui';
+
+import { useAuth } from '../../../internal';
+import { View, Flex } from '../../../primitives';
+import {
+  DefaultCurrentPassword,
+  DefaultError,
+  DefaultNewPassword,
+  DefaultSubmitButton,
+} from './defaultComponents';
+import { ChangePasswordProps } from './types';
+import { FormValues } from '../types';
+
+const logger = new Logger('ChangePassword');
+
+const ChangePassword: React.ComponentType<ChangePasswordProps> = (props) => {
+  const { onSuccess, onError } = props;
+  const [error, setError] = React.useState<string>(null);
+  const [formValues, setFormValues] = React.useState<FormValues>({});
+
+  const { user, isLoading: isAuthUserLoading } = useAuth();
+
+  /** Translations */
+  // TODO: add AccountSettingsTextUtil to collect these strings
+  const currentPasswordLabel = translate('Current Password');
+  const newPasswordLabel = translate('New Password');
+  const updatePasswordText = translate('Update password');
+
+  /** Auth API Handler */
+  const handleChangePassword = React.useCallback(async () => {
+    const { currentPassword, newPassword } = formValues;
+    try {
+      await changePassword({ user, currentPassword, newPassword });
+      setError(null);
+
+      if (onSuccess) {
+        onSuccess(); // notify success to the parent
+      }
+    } catch (e) {
+      const error = e as Error;
+      setError(error.message);
+
+      if (onError) {
+        onError(error); // notify error to the parent
+      }
+    }
+  }, [formValues, user, onSuccess, onError]);
+
+  /** Event Handlers */
+  const handleChange = React.useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      event.preventDefault();
+
+      const { name, value } = event.target;
+      setFormValues({ ...formValues, [name]: value });
+    },
+    [formValues]
+  );
+
+  const handleSubmit = React.useCallback(
+    (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+      handleChangePassword();
+    },
+    [handleChangePassword]
+  );
+
+  // return null if Auth.getCurrentAuthenticatedUser is still in progress
+  if (isAuthUserLoading) {
+    return null;
+  }
+
+  if (!user) {
+    logger.warn('<ChangePassword /> requires user to be authenticated.');
+    return null;
+  }
```

Comment:
Should we move these up to return earlier in the function?

Reply:
They have to be below the hooks to guarantee the hooks are run in the same order on each render (and not conditionally)

Reply:
Good point, I'll adjust accordingly.

Reply:
Actually, these have to be placed below all the `React.useCallback` hooks due to rules of hooks. 

I'll keep it here, but clarify comments around these return codes.

Reply:
---
After addressing Caleb's feedbacks, I ended up removing these `useCallback` hooks, so I moved these returns to the top.